### PR TITLE
Autogen handlers

### PR DESCRIPTION
**Pull request by Calebjh**
Object level C/R/U/D barebones GORM. No service/RPC direct integration just yet.

No end-to-end test just yet, but basic associations may work already, so long as they comply to GORM conventions (foreign key named as foreignTypeID), or are explicitly gorm tagged.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,33 +33,50 @@ make
...
 
-You can run `make install` or `go install` directly.
+##### Installation
+
+To use this tool, install it from code with `make install`, `go install` directly,
+or `go get github.com/infobloxopen/protoc-gen-gorm`.
+
+### Usage
+
+Once installed, the `--gorm-out=.` or `--gorm-out=${GOPATH}src`
```
Comment (johnbelamaric) [inline README.md]: s/gorm-out/gorm_out/

File: README.md [hunk 2]
```diff
@@ -31,33 +33,50 @@ make
...
+option can be specified in a protoc command to generate the .pb.gorm.go files.
+
+Any types with the `option (orm.opts).ormable = true` will have the following
```
Comment (johnbelamaric) [inline README.md]: Change `orm` to `gorm`, since this is really a gORM-specific plugin.

File: example/contacts/contacts.proto [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+message GetRequest {
+    uint64 id = 1;
+}
+
+message SMSRequest {
+    uint64 id = 1;
+    string message = 2;
+}
+
+service Contacts {
```
Comment (johnbelamaric) [inline example/contacts/contacts.proto]: This is ok for now, I think though that we want to nail down the conventions. @dmacthedestroyer suggested CreateRequest, for example. So, ultimately this example should follow those conventions (or we should just have the one example app and not put all the examples in here at all).

Something like:

```
message Contact {
...
}

message CreateRequest {
  Contact contact = 1;
  // maybe some other control or metadata, for example, a certain app may want to add
  // options to the create
  bool overwrite = 2; // overwrite an existing contact with the same email address
  // or add tags
  TagSet tags = 3;
}

message CreateResponse {
  Contact contact = 1;
}

service Contacts {
...
```

We need to have a separate discussion about what we want those basic calls to look like. They should be built in a way that they are extensible and integrate with the other Atlas services.

File: glide.lock [hunk 4]
```diff
@@ -1,5 +1,5 @@
 hash: 5d4fb27f861fb4a53f3cfafc8916cba71c7038b3438b57c24ffa36f028689cb2
-updated: 2018-03-06T17:28:16.140271-08:00
+updated: 2018-03-14T14:32:50.982036-07:00
```
Comment (johnbelamaric) [inline glide.lock]: Please move to dep in a separate PR (I created an issue for this).

File: example/contacts/contacts.pb.gorm.go [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+type ContactORM struct {
+	ID           uint64
+	FirstName    string
+	MiddleName   string
+	LastName     string
+	EmailAddress string
+}
+
+func (ContactORM) TableName() string {
+	return "Contact"
```
Comment (johnbelamaric) [inline example/contacts/contacts.pb.gorm.go]: I think the gORM conventions would make this `contacts`, wouldn't it?

File: README.md (hunk 2 above)
Comment (Calebjh) [reply README.md]: Should ormable -> gormable, too?

Comment (johnbelamaric) [reply README.md]: if the overall "orm" is changed to "gorm" then no, that's not needed.

File: glide.lock (hunk 4 above)
Comment (Calebjh) [reply glide.lock]: Will do
